app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -277,6 +287,10 @@ public void onClick(View v) {
         if (!isUsingActionBar()) {
             searchbox.setText(lastQueryString);
         }
+
+        if (!this.containsHereFunction) {
+            hereFunctionHandler.refreshLocation();
+        }

```

Comment:
Might be missing something but I feel like `refreshLocation` should be called when `containsHereFunction` is `true`, not `false`.